lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,6 +82,15 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
+		if (!isset($_SERVER['PHP_AUTH_USER'])) {
+			$httpAuthorization = $request->header('Authorization');
+			if ($httpAuthorization !== false && strlen($httpAuthorization) > 0 && strpos($httpAuthorization, 'basic') !== false) {
+				list($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW']) = explode(':', base64_decode(substr($httpAuthorization, 6)));
```

Comment:
Could we avoid changing `$_SERVER`? If lines 94-95 were moved above this new block, you'd be able to only modify the local `$username` and `$pass` variables.